src/controller/python/ChipDeviceController-ScriptBinding.cpp

```diff
@@ -555,6 +580,46 @@ PyChipError pychip_DeviceController_SetCheckMatchingFabric(bool check)
     return ToPyChipError(CHIP_NO_ERROR);
 }
 
+PyChipError pychip_DeviceController_SetIcdRegistrationParameters(chip::Controller::DeviceCommissioner * devCtrl, bool enabled,
+                                                                 uint8_t * icdSymmetricKeyOrNull, uint64_t icdCheckInNodeIdOrZero,
+                                                                 uint64_t icdMonitoredSubjectOrZero, uint32_t icdStayActiveMsec)
+{
+    if (!enabled)
+    {
+        return ToPyChipError(CHIP_NO_ERROR);
+    }
+
+    sCommissioningParameters.SetICDRegistrationStrategy(ICDRegistrationStrategy::kBeforeComplete);
+
+    if (icdSymmetricKeyOrNull != nullptr)
+    {
+        memcpy(sICDSymmetricKey, icdSymmetricKeyOrNull, sizeof(sICDSymmetricKey));
+    }
+    else
+    {
+        chip::Crypto::DRBG_get_bytes(sICDSymmetricKey, sizeof(sICDSymmetricKey));
+    }
+    if (icdCheckInNodeIdOrZero == 0)
+    {
+        icdCheckInNodeIdOrZero = devCtrl->GetNodeId();
+    }
+    if (icdMonitoredSubjectOrZero == 0)
+    {
+        icdMonitoredSubjectOrZero = icdCheckInNodeIdOrZero;
+    }
+    // These Optionals must have values now.
+    // The commissioner will verify these values.
+    sCommissioningParameters.SetICDSymmetricKey(ByteSpan(sICDSymmetricKey));
```

Comment:
Can we avoid accessing this global state? Does the `DeviceCommissioner` have setter/getters for this?